### ISSUE TITLE
Assertion error on `assert jnp.isfinite(x).all()`

### ISSUE THREAD

**Issue by sagitaninta** [jthlab/phlash#32]
I was trying to fit phlash in different regions of the chromosome of my own data set. As my vcf already filtered for all the problematic sites, the region parameter does not matter, but it asks me for a region so I specify the entire chromosome (e.g., ` chr23_c = phlash.contig(chr23_path, samples=["DlunBCh_2230"], region="23:1-48866549")` to run the entire chromosome 23). 

However it barely runs when I specify the entire chromosome as shown below

<img width="1254" height="947" alt="Image" src="https://github.com/user-attachments/assets/67ed20be-3976-4ac3-bb01-1c36deddfba5" />

Still the same when I specify half the size of it

<img width="1253" height="956" alt="Image" src="https://github.com/user-attachments/assets/3502d0ac-47d3-4b3f-a736-b6548cb21e3d" />

Run quite well when I specify only 1 Mbp in the region, but then end abruptly at `assert jnp.isfinite(x).all()`
<img width="1256" height="393" alt="Image" src="https://github.com/user-attachments/assets/0cde8e5c-d519-4694-9033-764fb739a42f" />

And similarly at 5 Mbp but faster (in ending haha)

<img width="1255" height="401" alt="Image" src="https://github.com/user-attachments/assets/0ae7cc4f-d4e9-4aca-9f2c-5cd684c259bd" />

Would you know what cause this behavior? Does it not like missing data? How should we better feed the region if the VCF is already filtered for all the SNPs that should be masked?

Many thanks in advance for your time and kind assistance!
